source/common/common/token_bucket_impl.h

```diff
@@ -28,9 +28,10 @@ class TokenBucketImpl : public TokenBucket {
   void maybeReset(uint64_t num_tokens) override;
 
 private:
-  const double max_tokens_;
-  const double fill_rate_;
-  double tokens_;
+  const uint64_t max_tokens_;
+  const uint64_t fill_rate_;
+  uint64_t tokens_;
+  uint64_t last_tokens_{0};
```

Comment:
CI is showing some test failures that seem related to this PR.

Example:
```
2021-05-11T10:26:38.4763974Z [ RUN      ] SharedTokenBucketImplTest.SynchronizedConsumeAndNextToken
2021-05-11T10:26:38.4764375Z TestRandomGenerator running with seed 812570616
2021-05-11T10:26:38.4764772Z test/common/common/shared_token_bucket_impl_test.cc:171: Failure
2021-05-11T10:26:38.4765176Z Expected equality of these values:
2021-05-11T10:26:38.4765732Z   std::chrono::milliseconds(200)
2021-05-11T10:26:38.4766446Z     Which is: 8-byte object <C8-00 00-00 00-00 00-00>
2021-05-11T10:26:38.4766875Z   token_bucket.nextTokenAvailable()
2021-05-11T10:26:38.4767613Z     Which is: 8-byte object <00-00 00-00 00-00 00-00>
2021-05-11T10:26:38.4768045Z Stack trace:
2021-05-11T10:26:38.4768624Z   0x534c5e: Envoy::SharedTokenBucketImplTest_SynchronizedConsumeAndNextToken_Test::TestBody()
2021-05-11T10:26:38.4769164Z   0xb93ec8: testing::internal::HandleExceptionsInMethodIfSupported<>()
2021-05-11T10:26:38.4769555Z   0xb93df5: testing::Test::Run()
2021-05-11T10:26:38.4769877Z   0xb94990: testing::TestInfo::Run()
2021-05-11T10:26:38.4770206Z ... Google Test internal frames ...
2021-05-11T10:26:38.4770385Z 
2021-05-11T10:26:38.4770690Z test/common/common/shared_token_bucket_impl_test.cc:185: Failure
2021-05-11T10:26:38.4771076Z Expected equality of these values:
2021-05-11T10:26:38.4771397Z   time_to_next_token.count()
2021-05-11T10:26:38.4771695Z     Which is: 0
2021-05-11T10:26:38.4771953Z   200
2021-05-11T10:26:38.4772205Z Stack trace:
```

Reply:
Sorry, I will check it and fix